### PR TITLE
[Feature] Init auditing functionality

### PR DESCRIPTION
**Pull request by Orrison**

### CODE REVIEW

File: app-modules/case/database/migrations/2023_03_06_000008_create_case_items_table.php [hunk 1]
```diff
@@ -6,21 +6,21 @@
...
             $table->string('respondent_id')->nullable();
             $table->longText('close_details')->nullable();
             $table->longText('res_details')->nullable();
-            $table->unsignedBigInteger('institution_id')->nullable();
-            $table->unsignedBigInteger('status_id')->nullable();
-            $table->unsignedBigInteger('type_id')->nullable();
-            $table->unsignedBigInteger('priority_id')->nullable();
-            $table->unsignedBigInteger('assigned_to_id')->nullable();
-            $table->unsignedBigInteger('created_by_id')->nullable();
+            $table->uuid('institution_id')->nullable();
```
Comment (dgoetzit) [inline app-modules/case/database/migrations/2023_03_06_000008_create_case_items_table.php:18]: We can probably do this in another PR, but makes sense to use `foreignUuid()` here and remove the extra migration to define the foreign key constraints. Will just need to re-order migrations.

File: app-modules/case/src/Models/CaseItem.php [hunk 2]
```diff
@@ -71,10 +75,12 @@
  *
  * @mixin Eloquent
  */
-class CaseItem extends BaseModel
+class CaseItem extends BaseModel implements Auditable
 {
     use SoftDeletes;
     use PowerJoins;
+    use AuditableTrait;
+    use HasUuids;
```
Comment (dgoetzit) [inline app-modules/case/src/Models/CaseItem.php:83]: We should probably just add the `HasUuids` trait to the `BaseModel`.

File: composer.json [hunk 3]
```diff
@@ -30,6 +31,7 @@
         "laravel/ui": "^4.2",
         "league/flysystem-aws-s3-v3": "^3.0",
         "livewire/livewire": "^3.0@beta",
+        "owen-it/laravel-auditing": "^13.5",
```
Comment (dgoetzit) [inline composer.json:34]: Should this be installed on the module instead of the root?

File: app-modules/prospect/src/Models/ProspectSource.php [hunk 4]
```diff
@@ -38,10 +42,11 @@
  *
  * @mixin Eloquent
  */
-class ProspectSource extends BaseModel
+class ProspectSource extends BaseModel implements Auditable
 {
     use HasUuids;
     use SoftDeletes;
+    use AuditableTrait;
```
Comment (dgoetzit) [inline app-modules/prospect/src/Models/ProspectSource.php:49]: We should probably add this trait to the `BaseModel` as well.

**Review by dgoetzit — APPROVED** [review]
This looks good. I left a few comments/suggestions/questions, but feel free to merge as they are non-blocking and can be addressed in a later PR.

File: app-modules/prospect/src/Models/ProspectSource.php (hunk 4 above)
Comment (Orrison) [reply app-modules/prospect/src/Models/ProspectSource.php:49]: I don't think we can at this point. Because the ADM Models extend the `BaseModel` and they don't have this trait

File: app-modules/case/database/migrations/2023_03_06_000008_create_case_items_table.php (hunk 1 above)
Comment (Orrison) [reply app-modules/case/database/migrations/2023_03_06_000008_create_case_items_table.php:18]: Yeah I can rearrange these at a later PR

File: composer.json (hunk 3 above)
Comment (Orrison) [reply composer.json:34]: Hmm, that should probably work and probably be how it works. Let me try

**Review by dgoetzit — APPROVED** [review]
